http-marshallers-java/http-jackson/src/main/java/org/apache/pekko/http/javadsl/marshallers/jackson/Jackson.java

```diff
@@ -113,7 +116,28 @@ static ObjectMapper createMapper(final Config config) {
         JsonFactory.builder()
             .streamReadConstraints(streamReadConstraints)
             .streamWriteConstraints(streamWriteConstraints)
+            .recyclerPool(getBufferRecyclerPool(config))
             .build();
     return new JsonMapper(jsonFactory);
   }
+
+  private static RecyclerPool<BufferRecycler> getBufferRecyclerPool(final Config cfg) {
+    switch (cfg.getString("buffer-recycler.pool-instance")) {
```

Comment:
As it's been used in the exceptions, how about extract to a local val？

Reply:
And maybe we can assert it should be null too.

Reply:
it is defaulted in reference.conf - virtually every config would blow up if we didn't have defaults - nowhere do we need to check for them not being set - because we have defaults in reference.conf